hyperapp.d.ts

```diff
@@ -0,0 +1,149 @@
+declare module "hyperapp" {
+  export interface VNodeData {
+    [key: string]: any
+  }
+
+  export interface VNode {
+    tag: string
+    data: VNodeData
+    children: string | VNode | Array<VNode>
+  }
+
+  export type viewFunction<State, Actions> = (
+    state: State,
+    actions: Actions
+  ) => VNode
+  export type emitFunction<Data> = (event: string, data?: Data) => Data
+  export type vnodeFunction = (
+    props: VNodeData,
+    children: string | VNode | Array<VNode>
+  ) => VNode
+
+  export interface Actions<State> {
+    [key: string]:
+      | ((
+          state?: State,
+          actions?: Actions<State>,
+          data?: any,
+          emit?: emitFunction<any>
+        ) => Partial<State> | any)
```

Comment:
I don't understand the `never`, to my knowledge it's used for functions that do not return at all (e.g. they only `throw`). 
Promises may not return a partial state, they can just call other actions and do not return a value. So I am for:
```js
=> Partial<State> | Promise<Partial<State>|void|null> | void | null
```